src/decision_model.jl

```diff
@@ -308,15 +332,290 @@ function LocalDecisionStrategy(d::Node, z::Array{VariableRef})
+- `diagram::InfluenceDiagram`: Influence diagram structure.
+- `z::OrderedDict{Name, DecisionVariable}`: Decision variables from `DecisionVariables` function.
+
+# Examples
+```julia
+μVars = cluster_variables_and_constraints(model, diagram, z)
+```
+"""
+
+function cluster_variables_and_constraints(model::Model, diagram::InfluenceDiagram, z::OrderedDict{Name, DecisionVariable})
+    # Get the RJT structure
+    # SHOULD WE DEFINE C_rjt and A_rjt AS PART OF THE INFLUENCE DIAGRAM STRUCT OR AS THEIR OWN, MAYBE BETTER TO DEFINE AS PART OF INFLUENCE DIAGRAM IF THERE IS NEVER A NEED
+    # TO HAVE AN INDEPENDENT RJT NOT LINKED TO AN INFLUENCE DIAGRAM
+    diagram.C_rjt, diagram.A_rjt = ID_to_RJT(diagram)
+
+    #μ- AND μ_bar-VARIABLES IN THE SAME LOOP, OTHERWISE NOT A PROBLEM, BUT DIFFERENT ORDER COMPARED TO PAPER AND ALSO SLIGHTLY WEIRD ORDER OF EQUATIONS IN MODEL PRINTOUT
+    # Variables corresponding to the nodes in the RJT
+    μVars = Dict{Name, μVariable}()
+    # Variables μ_{\bar{C}_v} = ∑_{x_v} μ_{C_v}
+    μ_barVars = Dict{Name, μVariable}()
+    for name in union(keys(diagram.C), keys(diagram.D))
+    #for name in union(keys(diagram.C), keys(diagram.D), keys(diagram.V))
+        μVars[name] = μVariable(name, μ_variable(model, name, diagram.States, diagram.C_rjt))
+        μ_barVars[name] = μVariable(name, μ_bar_variable(model, name, diagram.States, diagram.C_rjt, μVars[name].statevars))
+    end
+
+    # Enforcing local consistency between clusters, meaning that for a pair of adjacent clusters, 
+    # the marginal distribution for nodes that are included in both, must be the same.
+    for arc in diagram.A_rjt
+        if !isa(diagram.Nodes[arc[2]], ValueNode)
+            local_consistency_constraint(model, arc, diagram.C_rjt, μVars)
+        end
+    end
+
+    # Add in the conditional probabilities and decision strategies
+    # In our structure, value nodes are not stochastic and the objective function doesn't really work in this context. 
+    # However, adding a chance node representing stochasticity before the value node is a possibility.
+    for name in union(keys(diagram.C), keys(diagram.D))
+        factorization_constraints(model, diagram, name, μVars[name].statevars, μ_barVars[name].statevars, z)
+    end
+    return μVars
+end
+
+"""
+    RJT_objective(model::Model, diagram::InfluenceDiagram, μVars::Dict{Name, μVariable})
+
+Construct the RJT objective function.
+
+# Arguments
+- `model::Model`: JuMP model into which variables are added.
+- `diagram::InfluenceDiagram`: Influence diagram structure.
+- `μVars::Dict{Name, μVariable}`: Vector of moments.
+
+# Examples
+```julia
+RJT_objective(model, diagram, μVars)
 ```
 """
-function DecisionStrategy(z::DecisionVariables)
-    DecisionStrategy(z.D, z.I_d, [LocalDecisionStrategy(d, z_var) for (d, z_var) in zip(z.D, z.z)])
+
+function RJT_objective_function(model::Model, diagram::InfluenceDiagram, μVars::Dict{Name, μVariable})
```

Comment:
I think the other function names are nice and descriptive, but this one definitely should have expected value in it

Reply:
Changed to RJT_expected_value